taskgraph/Task.py

```diff
@@ -1162,20 +1228,55 @@ def join(self, timeout=None):
         LOGGER.debug(
             "joining %s done executing: %s", self.task_name,
             self.task_done_executing_event)
-        timed_out = self.task_done_executing_event.wait(timeout)
+        successful_wait = self.task_done_executing_event.wait(timeout)
         if self.exception_object:
             raise self.exception_object
-        return timed_out
+        return successful_wait
+
+    def get(self, timeout=None):
+        """Return the result of the `func` once it is ready.
+
+        If `timeout` is None, this call blocks until the task is complete
+        determined by a call to `.join()`. Otherwise will wait up to `timeout`
+        seconds before raising a `RuntimeError` if exceeded.
+
+        Parameters:
+            timeout (float): if not None this parameter is a floating point
+                number specifying a timeout for the operation in seconds.
+
+        Returns:
+            value of the result
+        Raises:
+            RuntimeError when `timeout` exceeded.
+
+        """
+        timeout = not self.join(timeout)
+        if timeout:
+            raise RuntimeError('call to get timed out')
+        return self._result
 
 
 def _get_file_stats(
-        base_value, hash_algorithm, ignore_list, ignore_directories):
+        base_value, hash_algorithm, ignore_list,
+        ignore_directories):
     """Return fingerprints of any filepaths in `base_value`.
 
     Parameters:
         base_value: any python value. Any file paths in `base_value`
             should be "os.path.norm"ed before this function is called.
             contains filepaths in any nested structure.
+        hash_algorithm (string): either a hash function id that
+            exists in hashlib.algorithms_available, 'exists', or
+            'sizetimestamp'. Any paths to actual files in the arguments will be
+            digested with this algorithm. If value is 'sizetimestamp' the
+            digest will only use the normed path, size, and timestamp of any
+            files found in the arguments. This value is used when
+            determining whether a task is precalculated or its target
+            files can be copied to an equivalent task. Note if
+            `hash_algorithm` is 'sizetimestamp' the task will require the
+            same base path files to determine equality. If it is a
+            `hashlib` algorithm only file contents will be considered. If this
+            value is 'exists' the value of the hash will be 'exists'.

```

Comment:
Yes, because this is `_get_file_stats`. Here I do want to sometimes `exists` hash the files. It's up to whomever calls it. I'm really second guessing this stuff James, but it has not been casually implemented!